pkg/resource/deploy/step.go

```diff
@@ -705,7 +708,15 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 	}
 
 	if outputs != nil {
-		s.new = resource.NewState(s.old.Type, s.old.URN, s.old.Custom, s.old.Delete, s.old.ID, inputs, outputs,
+		// There is a chance that the ID has changed. We want to allow this change to happen
+		// it will have changed already in the outputs, but we need to persist this change
+		// at a state level because the Id
+		if refreshed.ID.String() != "" && refreshed.ID != resourceID {
```

Comment:
I think we can just compare against the empty string directly:

```suggestion
		if refreshed.ID != "" && refreshed.ID != resourceID {
```

Reply:
I'm a bit surprised that we need the check against the empty string at all, to be honest. I suppose this is to guard against poorly-behaved providers that don't return an ID from `Read`?

Reply:
@pgavlin exactly that - without this, a lot of our tests were failing locally as it was changing the ID to an empty string and then wasn't able to do anything with it